src/firestore/api.ts

```diff
@@ -596,18 +629,112 @@ export class FirestoreIndexes {
     return result;
   }
 
+  /**
+   * List all databases that exist on a given project.
+   * @param project the Firebase project id.
+   */
+  async listDatabases(project: string): Promise<types.DatabaseResp[]> {
+    const url = `/projects/${project}/databases`;
+    const res = await this.apiClient.get<{ databases?: types.DatabaseResp[] }>(url);
+    const databases = res.body.databases;
+    if (!databases) {
+      return [];
+    }
+
+    return databases;
+  }
+
+  /**
+   * List all databases that exist on a given project.
+   * @param project the Firebase project id.
+   */
+  async locations(project: string): Promise<types.Location[]> {
+    const url = `/projects/${project}/locations`;
+    const res = await this.apiClient.get<{ locations?: types.Location[] }>(url);
+    const locations = res.body.locations;
+    if (!locations) {
+      return [];
+    }
+
+    return locations;
+  }
+
+  /**
+   * Get info on a Firestore database.
+   * @param project the Firebase project id.
+   * @param databaseId the name of the Firestore Database
+   */
+  async getDatabase(project: string, databaseId: string): Promise<types.DatabaseResp> {
+    const url = `/projects/${project}/databases/${databaseId}`;
+    const res = await this.apiClient.get<types.DatabaseResp>(url);
+    const database = res.body;
+    if (!database) {
+      return Promise.reject();
```

Comment:
In async functions, you can just throw an error instead of returning Promise.reject(). In this case, its probably appropriate to throw a not found error